src/action/SummationAction.cpp

```diff
@@ -35,8 +35,13 @@ void SummationAction::performAction(double time)
   targetValues.setZero();
 
   for (const auto &sourceData : _sourceDataVector) {
-    targetValues += sourceData->values();
+    auto sourceStample = sourceData->getStamples().back();
+    PRECICE_ASSERT(math::equals(sourceStample.timestamp, time::Storage::WINDOW_END));
+    auto sourceDataValues = sourceStample.sample.values;
+    targetValues += sourceDataValues;
   }
+
+  _targetData->setSampleAtTime(time::Storage::WINDOW_END, _targetData->sample());
```

Comment:
For the summation action we have a general problem here: The summation action takes n source data and performs a reduction onto a single target data. Here, we might have n source data that live on incompatible meshes in time (imagine multi coupling). A possible solution to this problem is to perform a resampling of the individual time-meshes of the source data onto the time-mesh of the target data. Currently, I'm only applying the action to the sample at `time::Storage::WINDOW_END`, because this is the common synchronization point of all data.

Reply:
Done. [73e0cea](https://github.com/precice/precice/pull/1612/commits/73e0ceada27ee10a3cd01857bba0057057bee2b7)